operate/services/manage.py

```diff
@@ -684,30 +684,31 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
                 OnChainState.NON_EXISTENT,
                 OnChainState.PRE_REGISTRATION,
             ):
-                required_olas = (
-                    staking_params["min_staking_deposit"]
-                    + staking_params[
-                        "min_staking_deposit"
-                    ]  # operator security deposit = max agent bond
+                staking_token_requirements = self._compute_staking_token_requirements(
+                    service_config_id, chain, staking_params
                 )
             elif chain_data.on_chain_state == OnChainState.ACTIVE_REGISTRATION:
-                required_olas = staking_params["min_staking_deposit"]
+                staking_token_requirements = self._compute_staking_token_requirements(
+                    service_config_id, chain, staking_params
+                )
+                staking_token_requirements[staking_params["staking_token"]] = (
```

Comment:
it looks like we can avoid overwriting these requirement by just reading the on_chain_state also in the compute function, and having this logic there

Reply:
Yes and no.

The output of this method is the service asset requirements (native, tokens, dual tokens, etc) required to deploy a service from scratch.

On another method, we retrieve the bonded assets.

So, the requirements needed will be service_asset_requirements - bonded_assets

Reply:
I opted for this way to have separation between what's bonded and what's on the Safe. We can discuss.

Reply:
In other words, the following invariant must be satisfied:

```
master_safe_balance + bonded_assets >= onchain_asset_requirements

=

master_safe_balance >= onchain_asset_requirements - bonded_assets
```